app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -4,109 +4,151 @@ import android.content.Intent
 import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.NavigationCommand
 import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
 import com.wire.android.notification.WireNotificationManager
 import com.wire.android.util.deeplink.DeepLinkProcessor
 import com.wire.android.util.deeplink.DeepLinkResult
-import com.wire.android.util.dispatchers.DispatcherProvider
-import com.wire.android.util.extension.intervalFlow
 import com.wire.kalium.logic.configuration.GetServerConfigResult
 import com.wire.kalium.logic.configuration.GetServerConfigUseCase
 import com.wire.kalium.logic.configuration.ServerConfig
-import com.wire.kalium.logic.feature.auth.AuthSession
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.feature.session.CurrentSessionFlowUseCase
 import com.wire.kalium.logic.feature.session.CurrentSessionResult
 import com.wire.kalium.logic.feature.session.CurrentSessionUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.distinctUntilChanged
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.shareIn
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
-@OptIn(ExperimentalCoroutinesApi::class)
+@Suppress("LongParameterList")
+@OptIn(
+    ExperimentalCoroutinesApi::class,
+    ExperimentalMaterial3Api::class
+)
 @ExperimentalMaterial3Api
 @HiltViewModel
 class WireActivityViewModel @Inject constructor(
-    private val currentSessionUseCase: CurrentSessionUseCase,
+    currentSessionUseCase: CurrentSessionUseCase,
+    currentSessionFlow: CurrentSessionFlowUseCase,
     private val getServerConfigUseCase: GetServerConfigUseCase,
     private val deepLinkProcessor: DeepLinkProcessor,
     private val notificationManager: WireNotificationManager,
-    private val dispatchers: DispatcherProvider
+    private val navigationManager: NavigationManager
 ) : ViewModel() {
 
-    private val currentSession: AuthSession? = runBlocking {
-        return@runBlocking when (val result = currentSessionUseCase()) {
-            is CurrentSessionResult.Success -> result.authSession
-            else -> null
+    private val navigationArguments = mutableMapOf<String, Any>(SERVER_CONFIG_ARG to ServerConfig.DEFAULT)
+    private val isUserLoggedIn = currentSessionUseCase() is CurrentSessionResult.Success
+
+    init {
+        val userIdFlow = currentSessionFlow()
+            .map { result ->
+                if (result is CurrentSessionResult.Success) result.authSession.userId
+                else null
+            }
+            .distinctUntilChanged()
+            .shareIn(viewModelScope, SharingStarted.WhileSubscribed(), 1)
```

Comment:
Maybe extract this logic to a `subscribeToUserAuthenticationChanges()` or `listenToUserAuthenticationChanges()` function?

Reply:
IMO `userIdFlow` describes what it is better. There is no any subscription or listening. It's just a flow of userId. 
Do you suggest to move part of code into some `private fun` just to have less code here?